app/src/main/java/com/alexis/myanimecompanion/data/remote/MyAnimeListAPI.kt

```diff
@@ -0,0 +1,66 @@
+package com.alexis.myanimecompanion.data.remote
+
+import com.alexis.myanimecompanion.QueryFieldsBuilder
+import com.alexis.myanimecompanion.data.remote.models.*
+import retrofit2.http.*
+
+interface MyAnimeListAPI {
+    @GET("anime")
+    suspend fun search(
+        @Query("q") query: String? = null,
+        @Query("limit") limit: Int? = null,
+        @Query("offset") offset: Int? = null,
+        @Query("fields") fields: String? = null,
+    ): SearchResult
+
+    @GET("anime/{anime_id}")
+    suspend fun getAnimeDetails(
+        @Path("anime_id") animeId: Int,
+        @Query("fields") fields: String? = QueryFieldsBuilder.fieldsForAnimeDetails().done(),
+        @Header("Authentication") token: String? = null,
+    ): Details
+
+    @GET("users/{user_name}/animelist")
+    suspend fun getUserAnimeList(
+        @Path("user_name") username: String
+    ): UserAnimeList
+
+    @GET("users/@me")
+    suspend fun getUserProfile(
+        @Header("Authorization") accessToken: String
+    ): User
+
+    @DELETE("anime/{anime_id}/my_list_status")
+    suspend fun deleteAnime(
+        @Header("Authorization") accessToken: String,
+        @Path("anime_id") animeId: Int,
+    )
+
+    // TODO PATCH or PUT ? documentation ambiguous
+    @PATCH("anime/{anime_id}/my_list_status")
+    suspend fun updateAnimeStatus(
+        @Header("Authorization") accessToken: String,
+        @Path("anime_id") animeId: Int,
+        @Field("status") status: String? = null,
```

Comment:
Review optional parameters. Please don't use optional parameters if all calls pass all parameters

Reply:
The optional parameters are there for when the uses updates _only_ the status (watching, plan to watch, completed...), or _only_ the number of episodes watched, or _only_ the score.